templates/login.defs.erb

```diff
@@ -56,9 +56,8 @@ HUSHLOGIN_FILE  .hushlogin
 
 # *REQUIRED*: The default PATH settings, for superuser and normal users. (they are minimal, add the rest in the shell startup files)
 ENV_SUPATH  PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
-ENV_PATH  PATH=/usr/local/bin:/usr/bin:/bin<% if not @additional_user_paths.empty? %>:<%= @additional_user_paths %><% end %>
+ENV_PATH  PATH=/usr/local/bin:/usr/bin:/bin<% if not scope['os_hardening::extra_user_paths'].empty? %>:<%= scope['os_hardening::additional_user_paths'] %><% end %>
```

Comment:
This requires to have the login_defs class to be declared by the main class. This means that you forbid users to make use of the login_defs class directly.
In this case you should add the private check method to login_defs.pp:
```
if $caller_module_name != $module_name {
  fail('You try to access a  private class')
}
```